src/lib/s3.js

```diff
@@ -71,27 +72,30 @@ export const getObjectList = id => {
 };
 
 export const getFiles = async id => {
-  const { Contents } = await getObjectList(id);
+  const { objectKeys } = await getObjectsMetadata(id);
   const data = {};
 
-  if (Contents.length === 0) {
+  if (objectKeys.length === 0) {
     return {};
   }
 
-  for (const content of Contents) {
-    // checks if it is the index file and skips over it
-    if (content.Key.indexOf('dependencies') !== -1) {
-      continue;
-    }
-    const params = { Bucket, Key: content.Key };
+  for (const key of objectKeys) {
+    const params = { Bucket, Key: key };
     const { Body } = await s3.getObject(params).promise();
     const file = JSON.parse(Body.toString('utf-8'));
+
     Object.assign(data, file);
   }
   return data;
 };
 
-export const getProfileSavedData = async id => {
+export const getObjectsMetadata = async id => {
+  const params = { Bucket, Key: `${id}/dependencies.json` };
+  const { Body } = await s3.getObject(params).promise();
+  return JSON.parse(Body.toString('utf-8'));
+};
+
+export const getData = async id => {
```

Comment:
This is not clear what it's doing and looks out of place in the s3 lib. s3 should not import the data lib.

I would refactor that out and clarify the `data` variables.